api/lib/db/database.py

```diff
@@ -0,0 +1,72 @@
+from sqlalchemy import create_engine
+from sqlalchemy.sql import func as sqlfunc
+from sqlalchemy.sql import text
+from sqlalchemy.orm import sessionmaker
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy import Column, String, Integer, Boolean, TIMESTAMP, ForeignKeyConstraint, UniqueConstraint, CheckConstraint, Text
+
+Base = declarative_base()
+Session = None
+
+def init(dbURI):
+    # I don't think Engine will be garbage collected since we
+    #    bind it to Session. Engine should still have a reference
+    engine = create_engine(
+        dbURI,
+        echo=True,
+        connect_args={'check_same_thread': False}
+    )
+    
+    global Session
+    Session = sessionmaker(autocommit=False, autoflush=False, bind=engine)
+    Base.metadata.create_all(engine)
+
+
+def db_session():
+    if Session is None:
+        raise RuntimeError("DB Session Was never initialized!")
+
+    return Session()
+
+class Participant(Base):
+    __tablename__ = 'participant'
+
+    id=Column(Integer, primary_key=True, autoincrement=True)
+    discord_user_id=Column(String, unique=True)
+    discord_guild_id=Column(String, unique=True)
+    last_updated = Column(TIMESTAMP, nullable=False, server_default=sqlfunc.current_timestamp(), server_onupdate=text(f' ON UPDATE {sqlfunc.current_timestamp()}'))
+
+# BUG One could create 2 games per duo by inviting eachother to their own game
+class Game(Base):
```

Comment:
>why do there need to be 4 game/move classes instead of 2 (the 2 moves classes are nearly identical replicas and the 2 games classes could easily be merged)? Ideally, games and moves shouldn't be separated from each other either for reasons outlined in (5).

- Game is our current "hot-set" the one that is repeatedly being editted/updated by players
- GameArchive is for games that have been previously completed so we can aim to delete them in the future if necessary (or lambda them to a an S3 bucket idek)
- Move is seperated from Game to allow for infinite movement loops and movement counting (i.e. finding out who made the last via time method). 

- Move